gitops/generate_mock.go

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitops
+
+import (
+	"fmt"
+	gitopsv1alpha1 "github.com/redhat-developer/gitops-generator/api/v1alpha1"
+	gitops "github.com/redhat-developer/gitops-generator/pkg"
+	"github.com/redhat-developer/gitops-generator/pkg/testutils"
+	"github.com/spf13/afero"
+	"path/filepath"
+	"strings"
+)
+
+type MockGenerator struct {
+	Outputs  *testutils.OutputStack
+	Errors   *testutils.ErrorStack
+	Executed []testutils.Execution
+}
+
+func NewMockGenerator(outputs ...[]byte) *MockGenerator {
+	return &MockGenerator{
+		Outputs:  testutils.NewOutputs(outputs...),
+		Errors:   testutils.NewErrors(),
+		Executed: []testutils.Execution{},
+	}
+}
```

Comment:
curious, shouldnt a mock for this interface reside in gitops repo?

Reply:
I can add it there since it originally came from there in the first place.  What I didn't want was the mock implementation to reside there

Reply:
its fine to keep it here then

Reply:
It looks like it's a limitation.  I can't move it to the gitops repo and have the implementation in the HAS repo because receiver types can't be coming from an external package.  They have to be defined in the same package as the method according to this thread  https://stackoverflow.com/questions/36293654/why-can-methods-in-go-only-be-declared-on-types-defined-in-the-same-package